aten/src/ATen/TensorIterator.cpp

```diff
@@ -402,9 +426,11 @@ void TensorIteratorBase::compute_types(const TensorIteratorConfig& config) {
   }
 
   // Checks that either the computation type is computable or unneeded
-  TORCH_INTERNAL_ASSERT(!(has_different_input_dtypes && !config.promote_inputs_to_common_dtype_ &&
-                        (has_undefined_outputs || config.enforce_safe_casting_to_output_ ||
-                        config.cast_common_dtype_to_outputs_)));
+  TORCH_CHECK(
```

Comment:
Same here, no need to improve error message, internal assert should remain internal assert